applications/nrf5340_audio/src/bluetooth/bt_renderer/volume/bt_rend_vol.c

```diff
@@ -0,0 +1,369 @@
+{
+	if (err) {
+		LOG_WRN("VCS discover finished callback error: %d", err);
+	} else {
+		LOG_INF("VCS discover finished");
+	}
+}
+
+int bt_rend_vol_set(uint8_t volume)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
+
+		for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_set_vol(vcs_client_peer[i], volume);
+				if (ret) {
+					LOG_WRN("Failed to set volume for remote channel %d, ret = "
+						"%d",
+						i, ret);
+				}
+			}
+		}
+
+		return 0;
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_set_vol(volume);
+	}
+
+	LOG_WRN("VCP not enabled");
+	return -EIO;
+}
+
+int bt_rend_vol_up(void)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
+
+		for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_unmute_vol_up(vcs_client_peer[i]);
+				if (ret) {
+					LOG_WRN("Failed to volume up for remote channel %d, ret = "
+						"%d",
+						i, ret);
+				}
+			}
+		}
+
+		return 0;
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_unmute_vol_up();
+	}
+
+	LOG_WRN("VCP not enabled");
+	return -EIO;
+}
+
+int bt_rend_vol_down(void)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
+
+		for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_unmute_vol_down(vcs_client_peer[i]);
+				if (ret) {
+					LOG_WRN("Failed to volume down for remote channel %d, ret "
+						"= %d",
+						i, ret);
+				}
+			}
+
+			return 0;
+		}
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_unmute_vol_down();
+	}
+
+	LOG_WRN("VCP not enabled");
+	return -EIO;
+}
+
+int bt_rend_vol_mute(void)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
+
+		for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_mute(vcs_client_peer[i]);
+				if (ret) {
+					LOG_WRN("Failed to mute for remote channel %d, ret "
+						"= %d",
+						i, ret);
+				}
+			}
+		}
+
+		return 0;
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_mute();
+	}
+
+	LOG_WRN("VCP not enabled");
+	return -EIO;
+}
+
+int bt_rend_vol_unmute(void)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		int ret;
+
+		for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+			if (vcs_client_peer[i] != NULL) {
+				ret = bt_vcp_vol_ctlr_unmute(vcs_client_peer[i]);
+				if (ret) {
+					LOG_WRN("Failed to unmute for remote channel %d, "
+						"ret = %d",
+						i, ret);
+				}
+			}
+		}
+
+		return 0;
+	} else if (IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		return bt_vcp_vol_rend_unmute();
+	}
+
+	LOG_WRN("VCP not enabled");
+	return -EIO;
+}
+
+int bt_rend_vol_discover(struct bt_conn *conn)
+{
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+
+		int ret, index;
+
+		if (vcs_client_peer_exists(conn)) {
+			return -EAGAIN;
```

Comment:
Why -EAGAIN here?

Reply:
Because this vcs_client_peer already exists, so we shouldn't do discovery again